homeassistant/components/lg_soundbar/media_player.py

```diff
@@ -126,10 +126,15 @@ def update(self):
             if equaliser >= len(temescal.equalisers):
                 temescal.equalisers.append("unknown " + str(equaliser))
 
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
     @property
     def unique_id(self):
         """Return the device's unique ID."""
-        return self._uuid
+        return self._hostname
```

Comment:
Is there something better we can use?  This doesn't meet the unique id requirements either

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
The only information provided by the discovery is (with the UUID changing everytime):
```
{'host': '192.168.178.123', 'port': 9741, 'hostname': 'soundbar-domain.local.', 'properties': {'PROTOVER': '1.0', 'UUID': '23453245-k434-3454-8f76-77sdf897sdf89', 'Model_Type': '0', 'ASSISTANT': 'true'}}
```
The only other idea i have would be to use the MAC but for that we would need to get it somehow. Would it be ok to add additional dependency just to get the MAC?

Reply:
Getting the mac address from the ip address is unfortunately unreliable in some network setups. So we don't want to do that.

Reply:
Does the device require authentication? Can we get it to give a unique id using the existing package?

Reply:
There is `get_mac_info`/`MAC_INFO_DEV` but it does not work for me (message not found error). Maybe its only for development or for a different model. I didn't find any other method offering a unique identifier only settings, os version, model type, ...
Any other ideas i can test?

Reply:
A hostname could be unique if it contains a unique ID, however `soundbar-domain.local.` doesn't seem like it's unique at all?

Reply:
In this case we should use the config entry ID as the unique ID (once this integration uses a config flow)

Reply:
> A hostname could be unique if it contains a unique ID, however `soundbar-domain.local.` doesn't seem like it's unique at all?

I am not sure if there was a unique hostname before (ID), but `soundbar` is the name i gave it and `domain` is my local domain name (replaced it).

> In this case we should use the config entry ID as the unique ID (once this integration uses a config flow)

How would the config entry ID provide provide the matching / fix the multiple entities in case the IP changes? Don't we need something inside the config flow to determine if the discovered device is the same as before? I tried to implement the config flow before but got stuck on that part.

Reply:
Can the zeroconf discovery step be used to find matching config entry and update the host?

Reply:
I guess, but then we would have to match by hostname again (UUID changes every time and IP can change too). If its only about using a proper unique ID for the entity then i could live with that. But the matching wouldn't be any different than the PR right now.

Reply:
I suggest we remove the unique ID for now. Once this integration has been moved to be based around a config flow, we can base the unique ID on the config entry.

Reply:
I think this just moves the matching logic (discovered device to config flow) to the config flow and doesn't really address the root problem (no usable unique id in the discovery information). But it's fine with me as long as this fixes the duplicated devices problem.

Reply:
Ok, so let's drop unique ID and this PR can be merged.

Reply:
You mean removing
```
@property
def unique_id(self):
  return self._hostname
```
? Isn't this needed to get the same entity after a new discovery / hass restart (which in turn means duplicate entities if you wanna change the name)?

Reply:
Yes, that needs to be removed. 

It will then base the entity ID on the name of the entity. It will not be possible to change the entity ID.

Since the other unique ID is not unique, this is what should be done until a config flow is added and the config entry ID can be used.